internal/backend/backend.go

```diff
@@ -167,13 +179,41 @@ func (b *Backend) Close(ctx context.Context) error {
 	return nil
 }
 
-func (b *Backend) getUserID(username string, password []byte) (string, error) {
+func (b *Backend) getUserID(ctx context.Context, username string, password []byte) (string, error) {
+	if atomic.LoadInt32(&b.loginErrorCount) >= maxLoginAttempts {
+		b.lockLogin.Lock()
```

Comment:
When using atomics, you don't need to use the mutex.

Reply:
the lock is here to delayed the answer so that the IMAP client wont see that the login is blocked but rather the answer will take longer. (as Jakub mentionned)
here the purpose is really to wait for the timer to end before treating a new login attempt.

Reply:
I see. In that case there is still a small chance that two clients login in at the same time with the invalid credentials and one of them will reset the login count before the timer has a chance to start.

E.g.:

Client A  fails login and increases the login count to `maxLoginAttempts` and before it has a chance top reach the go routine that launches the timer client B also fails the login and acquires the mutex before the goroutine can. Now B resets the counter before the timer of A runs to completion.

Reply:
The proposed changes address the counter reset issue, but don't exactly force the client to wait until the timer has finished in the example listed above.

Reply:
True... always have to figure out how to handle this